.ci/Jenkinsfile

```diff
@@ -185,3 +246,37 @@ def notifyStatus(def args = [:]) {
                       subject: args.subject,
                       body: args.body)
 }
+
+def downloadSignedBinaries() {
+  googleStorageDownload(bucketUri: "${env.BUCKET_SIGNED_ARTIFACTS_PATH}/*",
+                        credentialsId: env.BUCKET_CREDENTIALS,
+                        localDirectory: "${DIST_FOLDER}/",
+                        pathPrefix: "${env.BUCKET_SUBFOLDER_SIGNED_ARTIFACTS}")
+  archiveArtifacts(allowEmptyArchive: false, artifacts: "${DIST_FOLDER}/*.*")
+}
+
+def uploadBinariesToBeSigned(extensions = [:]) {
+  extensions?.each { extension ->
+    googleStorageUpload(bucket: env.BUCKET_PATH,
+                        credentialsId: env.BUCKET_CREDENTIALS,
+                        pathPrefix: "${env.DIST_FOLDER}/",
+                        pattern: "${env.DIST_FOLDER}/*.${extension}",
+                        sharedPublicly: false,
+                        showInline: true)
+  }
+}
+
+def runReleaseWithKeychain() {
+  def vault = getVaultSecret(secret: 'secret/observability-team/ci/synthetics-recorder-macos-vault').data
+  if (!vault.containsKey('vault_secret_id') || !vault.containsKey('vault_role_id') || !vault.containsKey('vault_addr')) {
+    error("runReleaseWithKeychain: macos secret could not be accessed correctly")
+  }
+  def token = getVaultSecret.getVaultToken(vault.vault_addr, vault.vault_role_id, vault.vault_secret_id)
+  def apple = getVaultSecret.getVaultSecretObject(vault.vault_addr, 'secret/release/apple', token).data
+  if (!apple.containsKey('keychain-password')) {
+    error("runReleaseWithKeychain: apple secret could not be accessed correctly")
+  }
+  withEnvMask(vars: [[var: 'KEYCHAIN_PASSWORD', password: apple.get('keychain-password')]]){
```

Comment:
Should we omit `KEYCHAIN_PASSWORD` env var from [this line](https://github.com/elastic/synthetics-recorder/blob/main/scripts/before-pack.js#L40
) just to be secure? cc @justinkambic

Reply:
This is masked with https://github.com/elastic/apm-pipeline-library/blob/cf03a54281188247d61f2a5088e25c7fe61e7eb0/vars/withEnvMask.groovy#L19-L20

No issues with the proposal to omit it

Reply:
Would you mind if I ask you to contribute with the omitting change to this PR? You probably can do it much faster as I'm not fluent in this technology 🙏

Reply:
Sure thing - just pushed the commit.